test/nexus-integration.py

```diff
@@ -134,6 +135,63 @@ def test_search_nexus3_pyversion(self):
         )
         self.assertListEqual(versions, [pipio.Version('1.0.0'), pipio.Version('1.0.1rc1'), pipio.Version('1.0.1')])        
 
+class TestPut(unittest.TestCase):
+
+    def test_upload_package_for_pep440_compliant_version(self):
+        pkg_name = 'test_package2' # must differ from package names used in TestCheck
+        src_repo = os.path.join(THISDIR, 'generalized_package')
+        with tempfile.TemporaryDirectory() as tmpdir:
+            dst_repo = os.path.join(tmpdir, 'generalized_package')
+            shutil.copytree(src_repo, dst_repo)
+            rc = subprocess.run(['python', 'setup.py', 'sdist'],
+                check=True, cwd=dst_repo,
+                env={
+                    **os.environ,
+                    'TEST_PACKAGE_NAME': pkg_name,
+                    'TEST_PACKAGE_VERSION': '1.2.3.dev10+g123abc45',
+                }
+            )
+            print("sdist returned", rc)
+            out.out(
+                os.path.join(dst_repo, "dist"),
+                {
+                    'source': {
+                        'name': pkg_name,
+                        'repository': make_input(None)['source']['repository'],
+                    },
+                    'params': {'glob': '*.tar.gz'}
+                }
+            )
+
+    def test_fail_to_upload_if_package_version_not_pep440_compliant(self):
```

Comment:
There is the same test also in https://github.com/Pix4D/concourse-pypi-resource/blob/9d6525f8eaeafaef96d348ac181c99d256610270/test/pypi-integration.py#L96

but there I could not have a `test_upload_package_for_pep440_compliant_version` because it would upload it to PyPi and conflict with `test_upload_pypi`.